src/Compiler/SyntaxTree/SyntaxTree.fs

```diff
@@ -47,10 +47,14 @@ type SynLongIdent =
     member this.IdentsWithTrivia =
         let (SynLongIdent (lid, _, trivia)) = this
 
-        if lid.Length <> trivia.Length then
-            failwith "difference between idents and trivia"
-        else
+        if lid.Length = trivia.Length then
+            List.zip lid trivia |> List.map SynIdent
+        elif lid.Length > trivia.Length then
+            let delta = lid.Length - trivia.Length
+            let trivia = [ yield! trivia; yield! List.replicate delta None ]
             List.zip lid trivia |> List.map SynIdent
+        else
+            failwith "difference between idents and trivia"
```

Comment:
@auduchinok what would you do in this case? 
This is just plain wrong if this state is ever achieved.

Reply:
We can probably leave it as is if we aim to fix all places where these nodes are created.